kanidmd/src/lib/actors/v1_read.rs

```diff
@@ -167,6 +172,38 @@ impl QueryServerReadV1 {
         res
     }
 
+    pub async fn handle_online_backup(&self, msg: LiveBackupEvent, outpath: &str) {
+        let mut audit = AuditScope::new("online backup", msg.eventid, self.log_level);
+
+        ltrace!(audit, "Begin online backup event {:?}", msg.eventid);
+
+        // TODO: might be something like "backup-timestamp.json"
+        let now: DateTime<Utc> = Utc::now();
+        let timestamp = now.to_rfc3339_opts(SecondsFormat::Secs, true);
+        let dest_file = format!("{}/backup-{}.json", outpath, timestamp);
+
+        // TODO: handle the max versions to keep here?
+        let idms_prox_read = self.idms.proxy_read_async().await;
+        lperf_op_segment!(
+            &mut audit,
+            "actors::v1_read::handle<OnlineBackupEvent>",
+            || {
+                let res = idms_prox_read
+                    .qs_read
+                    .get_be_txn()
+                    .backup(&mut audit, &dest_file);
+
+                ladmin_info!(audit, "online backup result: {:?}", res);
+                #[allow(clippy::expect_used)]
+                res.expect("Online backup failed");
+            }
+        );
+        // At the end of the event we send it for logging.
```

Comment:
I'm thinking we could do the cleanup here but I'm not sure what's the safest way that "won't" accidentally remove something the admin wants to keep.

Reply:
I also think this is a good place. Well, we would always keep the minimum versions requested. I was thinking of some sort of `ls backup*json`, sort it and remove what is above the limit. Other ideas? First research brought up `glob` is this an option or just do a `fs::read_dir()`?

Reply:
I think it's better here to be very specific. Instead of a glob, we should list the dir and match a regex on the file names I think. 

IE if the admin moved backup.timestamp.json to backup.superimportant.json, the glob would still remove it. But a regex would match that it's no longer a timestamp so we don't remove it. 

Perhaps for the moment backup-rotation may be a second PR because it could be a fail bit of work on it's own. 

Another option is we write out a .backup.control.json file that contains metadata abotu the backups we have created ad what exists, and we can use that instead to know what files might exist that we need to remove/prune.

Reply:
Should be done.
We are using the regex now and do a very simple cleanup of versions by max count